util/movement-types/src/block.rs

```diff
@@ -134,6 +134,15 @@ impl Commitment {
 	}
 }
 
+impl fmt::Display for Commitment {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		for byte in &self.0 {
+			write!(f, "{:02x}", byte)?;
+		}
```

Comment:
I'm pretty sure there should be a non-allocating wrapper for hex formatting of byte slices, it's needed in so many places. Maybe one should be added to movement-types if there's nothing to grab off crates.io.